### PR TITLE
SONARPY-396 update custom-rule example with sonar-python API

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: python-custom-rules/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+        <configuration>
+          <skipTests>false</skipTests>
```
Comment (pynicolas) [inline python-custom-rules/pom.xml]: Why?

I have a maven warning related to the enclosing block:
```
[WARNING] Some problems were encountered while building the effective model for org.sonarsource.samples:python-custom-rules-plugin:sonar-plugin:1.0-SNAPSHOT
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-surefire-plugin is missing. @ line 44, column 15
```

File: python-custom-rules/pom.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      <groupId>org.sonarsource.python</groupId>
+      <artifactId>sonar-python-plugin</artifactId>
+      <type>sonar-plugin</type>
+      <version>2.0-SNAPSHOT</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.sonarsource.python</groupId>
+      <artifactId>python-checks-testkit</artifactId>
+      <version>2.0-SNAPSHOT</version>
```
Comment (pynicolas) [inline python-custom-rules/pom.xml]: We will need to update this version once 2.0 is released.

File: python-custom-rules/pom.xml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+          <skipTests>false</skipTests>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.sonarsource.sonar-packaging-maven-plugin</groupId>
+        <artifactId>sonar-packaging-maven-plugin</artifactId>
+        <version>1.18.0.372</version>
+        <extensions>true</extensions>
+        <configuration>
+          <pluginClass>org.sonar.samples.python.CustomPythonRulesPlugin</pluginClass>
```
Comment (pynicolas) [inline python-custom-rules/pom.xml]: [According to Julien](https://github.com/SonarSource/sonar-python/commit/ca010adf85d2b43024d92398272620d34790900b#r35539255), we should add `<requirePlugin>...` here. I suppose we can't do it before the release of SonarPython 2.0.

File: python-custom-rules/pom.xml (hunk 1 above)
Comment (benzonico) [reply python-custom-rules/pom.xml]: good catch : this is something that remains from when I made this a module of the project. Which is not the case anymore. So I need to fix it also in ITs.

File: python-custom-rules/pom.xml (hunk 2 above)
Comment (benzonico) [reply python-custom-rules/pom.xml]: sure.

File: python-custom-rules/pom.xml (hunk 3 above)
Comment (benzonico) [reply python-custom-rules/pom.xml]: Thanks, I'll update the IT as well.

**Pull request merged**